tests/src/test/java/com/maxio/advancedbilling/controllers/proformainvoices/ProformaInvoicesCreator.java

```diff
@@ -0,0 +1,295 @@
+package com.maxio.advancedbilling.controllers.proformainvoices;
+
+import com.maxio.advancedbilling.AdvancedBillingClient;
+import com.maxio.advancedbilling.TestClient;
+import com.maxio.advancedbilling.exceptions.ApiException;
+import com.maxio.advancedbilling.models.AllocateComponents;
+import com.maxio.advancedbilling.models.Component;
+import com.maxio.advancedbilling.models.CreateAllocation;
+import com.maxio.advancedbilling.models.CreateUsage;
+import com.maxio.advancedbilling.models.CreateUsageRequest;
+import com.maxio.advancedbilling.models.Customer;
+import com.maxio.advancedbilling.models.InvoiceAddress;
+import com.maxio.advancedbilling.models.InvoiceCustomer;
+import com.maxio.advancedbilling.models.InvoiceLineItem;
+import com.maxio.advancedbilling.models.InvoiceSeller;
+import com.maxio.advancedbilling.models.IssueServiceCredit;
+import com.maxio.advancedbilling.models.IssueServiceCreditRequest;
+import com.maxio.advancedbilling.models.Product;
+import com.maxio.advancedbilling.models.ProductFamily;
+import com.maxio.advancedbilling.models.ProformaInvoice;
+import com.maxio.advancedbilling.models.ProformaInvoiceCredit;
+import com.maxio.advancedbilling.models.Subscription;
+import com.maxio.advancedbilling.models.containers.CreateUsageComponentId;
+import com.maxio.advancedbilling.models.containers.IssueServiceCreditAmount;
+import com.maxio.advancedbilling.utils.TestSetup;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+
+import static com.maxio.advancedbilling.utils.TestFixtures.INVOICE_SELLER;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+public class ProformaInvoicesCreator {
```

Comment:
Alternatively you can create ProformaInvoiceBaseTest and extend test classes.

Reply:
Not all classes use it so I'll leave it as it is for now

Reply:
👍 , I'll consider adding it within the new tests